python/treelite/sklearn/exporter.py

```diff
@@ -0,0 +1,233 @@
+        if n_targets == 1:
+            subestimator_state["n_classes_"] = n_classes[0]
+        else:
+            subestimator_state["n_classes_"] = n_classes.tolist()
+    subestimator.__setstate__(subestimator_state)
+    return subestimator
+
+
+def export_model(model: Model):
+    """
+    Export as scikit-learn RandomForest or GradientBoosting.
+
+    Note
+    ----
+    Currently only random forests can be exported as scikit-learn model objects.
+    Support for gradient boosted trees and other kinds of tree models will be
+    added in the future.
+
+    Parameters
+    ----------
+    model : :py:class:`Model`
+        Treelite mobel to export
+
+    Returns
+    -------
+    sklearn_model : object of type \
+                    :py:class:`~sklearn.ensemble.RandomForestRegressor` / \
+                    :py:class:`~sklearn.ensemble.RandomForestClassifier` / \
+                    :py:class:`~sklearn.ensemble.GradientBoostingRegressor` / \
+                    :py:class:`~sklearn.ensemble.GradientBoostingClassifier`
+        Scikit-learn model
+    """
+    # pylint: disable=too-many-locals
+    try:
+        from sklearn import __version__ as sklearn_version
+        from sklearn.ensemble import RandomForestRegressor
+    except ImportError as e:
+        raise TreeliteError("This function requires scikit-learn package") from e
+
+    header_accessor = model.get_header_accessor()
+    average_tree_output = (
+        _ensure_scalar_int(header_accessor.get_field("average_tree_output")) == 1
+    )
+    n_features = _ensure_scalar_int(header_accessor.get_field("num_feature"))
+    n_trees = _ensure_scalar_int(header_accessor.get_field("num_tree"))
+    n_targets = _ensure_scalar_int(header_accessor.get_field("num_target"))
+    n_classes = _ensure_numpy(header_accessor.get_field("num_class"))
+    leaf_vector_shape = _ensure_numpy(header_accessor.get_field("leaf_vector_shape"))
+    target_id = _ensure_numpy(header_accessor.get_field("target_id"))
+    class_id = _ensure_numpy(header_accessor.get_field("class_id"))
+    tree_depths = model.get_tree_depth()
+
+    # Heuristics to ensure that the model can be represented as scikit-learn random forest
+    # 1. average_tree_output must be True
+    # 2. n_classes[i] must be identical for all targets
+    # 3. Each leaf must yield an output of shape (n_targets, n_classes)
+    # 4. target_id[i] must be either 0 or -1
+    # 5. class_id[i] must be either 0 or -1
+    def raise_not_rf_error():
+        raise NotImplementedError(
+            "Currently only random forests can be exported as scikit-learn model objects. "
+            "Gradient boosting models and other kinds of decision tree models are not yet "
+            "supported."
+        )
+
+    if not average_tree_output:
+        raise_not_rf_error()
+    if not np.all(n_classes == n_classes[0]):
+        raise_not_rf_error()
+    if not np.array_equal(leaf_vector_shape, [n_targets, n_classes.max()]):
```

Comment:
I guess it is the same here? You could have a RF but one of these conditions isn't true, in which case the "only RFs are supported" error message would be confusing.

Reply:
See my remark above. For all scikit-learn RF, this condition holds true.